WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -91,48 +88,35 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onStop() {
-        mDispatcher.unregister(this);
-        super.onStop();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResult();
+        }
+        finishAndGoBackToSource();
     }
 
     @Override
-    protected void onStart() {
-        super.onStart();
-        mDispatcher.register(this);
+    public void onBackPressed() {
+        super.onBackPressed();
+        finishAndGoBackToSource();
     }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && mSource == Source.STATS) {
-            showStats();
+    private void trackResult() {
+        if (!TextUtils.isEmpty(mReason)) {
+            AppLog.e(AppLog.T.API, "Could not connect to Jetpack, reason: " + mReason);
+            ToastUtils.showToast(this, mReason);
         } else {
-            finish();
+            trackWithSource(Stat.SIGNED_INTO_JETPACK, mSource);
```

Comment:
I am not really sure what this actually tracks. It really doesn't seem like it belongs here though, can you provide a bit more context about what this stat is and when it should be tracked?

Reply:
I don't really have much more context. We're sending (according to requirements) two events - install_jetpack_completed and signed_into_jetpack. I'm sending the first one when we trigged the deeplink activity (when the flow is finished and we get back to the app) and the second one after we determine that the user is logged in in the app. I didn't get any specifics as to when the events are supposed to be sent and this seemed to make the most sense. I'm open to suggestions.